### ISSUE TITLE
[concourse/stable] External worker without team association

### ISSUE THREAD

**Issue by aegershman** [helm/charts#14869]
**Is your feature request related to a problem? Please describe.**

I have external workers in multiple environments (Azure, vSphere, etc.) all phoning into a central `web` deployment. But using external workers requires they be associated to a team. That is, your external worker configuration must have a `concourse.worker.team` on it:

```yml
concourse:
  worker:
   # team is required
    team: main
    tag: tag-youre-it,guten-tag,another-tag
    tsa:
      host: concourse.sandbox.aegershman.com
```

I'm under the impression this limits the worker to only be usable by the `main` team. 

**Describe the solution you'd like**
I'd like to wire up external workers without requiring to specify a `team`, and instead let them be associated to _any_ team. Their usage will be driven by `tag`. That way I could have multiple Concourse teams for different environments. For example, I could use individual Concourse teams for `aws-us-west-2-sandbox`, `aws-us-west-2-dev`, etc., and they could all use an external worker pool tagged with `aws-us-west-2`.

I don't believe this is a limitation of Concourse(?), I think it's because we're currently assuming any external worker added to a Concourse cluster will be reserved for a specific team. Thus to add an external worker, [it must be done under `team-authorized-keys`](https://github.com/helm/charts/blob/master/stable/concourse/templates/web-secrets.yaml#L99-L100) and thus require a `team` association.

Haven't tested this at all, but perhaps the `team:` name requirement in `teamAuthorizedKeys` could be modified so it doesn't _require_ the worker to have a team association, and instead be seen as an an external worker that's accessible for `any` team to use? https://github.com/helm/charts/blob/master/stable/concourse/values.yaml#L1726-L1728

**Additional context**
Some day it would be even better if workers could be added to multiple teams (https://github.com/concourse/concourse/issues/2633), but that's an issue for a different day.

Thoughts?

Thanks for your time and consideration!

**Comment by cirocosta** [comment]
Hey @aegershman,

> But using external workers requires they be associated to a team.

oh, that is actually not true 🤔  Did you have trouble when not having a `team` specified? in [`hush-house`](https://github.com/concourse/hush-house) we have a pool of "external" workers that have no team associated with them: https://github.com/concourse/hush-house/blob/41e3b35381b90c21aefcfc22ba9225b6d289bffe/deployments/with-creds/worker/values.yaml

(even though that `values.yaml` doesn't contain the credentials - we keep an extra `.values.yaml` in LastPass -, I double checked here are there's no use of `team authorized keys`)

Would mind sharing what problems you faced when trying to have external workers without teams associated with them?

Thanks!

**Comment by aegershman** [comment]
Hm I seeeee... So in the [`hush-house` deployment](https://github.com/concourse/hush-house/blob/41e3b35381b90c21aefcfc22ba9225b6d289bffe/deployments/with-creds/hush-house/values.yaml) (where `web` is hosted, without a pool of workers), do you have the worker's public key (`secrets.workerKeyPub`) configured somewhere hidden, such as `.values`? I'm under the impression the workers will need the [public key of the `host`](https://github.com/concourse/hush-house/blob/41e3b35381b90c21aefcfc22ba9225b6d289bffe/deployments/with-creds/worker/values.yaml#L53-L55), and the `host` (web) will need the [public key of the worker](https://github.com/helm/charts/blob/master/stable/concourse/values.yaml#L1865-L1866)

Is that the case?

Because if it _is_, then that's what we're successfully doing with one of our worker pools. But if you want to add more than one external worker pools (which aren't associated to a team, they'd be available to all), it appears there's no place to configure extras. In a `web` deployment, it appears you can only configure [one `workerKeyPub`](https://github.com/helm/charts/blob/master/stable/concourse/values.yaml#L1865-L1866), or you can configure multiple external worker pools, [but they must be associated to a team](https://github.com/helm/charts/blob/master/stable/concourse/values.yaml#L1743-L1754)

I suppose I could configure _all_ the external worker pools with the same private key so that when I configure the `workerKeyPub` on the `web`-only deployment, all the external worker pools authenticate, but it's not the most secure option.

---

tl;dr, let's say you wanted to add another external pool of workers (in a different namespace or cluster) in the hush-house project. What would your [web-only deployment (named hush-house 😉)](https://github.com/concourse/hush-house/blob/41e3b35381b90c21aefcfc22ba9225b6d289bffe/deployments/with-creds/hush-house/values.yaml) look like?

thanks a lot, I appreciate your time 👍

**Comment by aegershman** [comment]
Apologies if this is spammy but @cirocosta just a bit more research: It appears the [Concourse BOSH 4.2.1 deployment accepted an array of authorized worker keys under `tsa.authorizedKeys`](https://bosh.io/jobs/tsa?source=github.com/concourse/concourse&version=4.2.1#p%3dauthorized_keys). 

Right now it appears [the chart is configured for a single key under `CONCOURSE_TSA_AUTHORIZED_KEYS`](https://github.com/helm/charts/blob/master/stable/concourse/templates/web-deployment.yaml#L979-L980). 

Any thoughts? Would there be consideration for a PR that make changes which allow for multiple `authorizedKeys` from multiple external workers? This could be used for hosting workers on multiple clusters, multiple IaaSes, and for hosting a few Windows-VM workers.

Again, thanks a lot for your time & consideration

**Comment by aegershman** [comment]
Fixed by #15220

thanks very much 👍

**Comment by aegershman** [comment]
Wait a second, no it doesn't, I'm a dingus.

**Comment by cirocosta** [comment]
oh hahah

Sorry for the radio silence, I'll take a deeper look into it 😁  Thanks!

**Comment by aegershman** [comment]
No worries, all good, this isn't a high priority; I don't want to be naggy or throw off prioritizations, just wanted to report the use-case 👍

**Comment by cirocosta** [comment]
🙌
